plugins/output/animationAssetUnpack/index.js

```diff
@@ -0,0 +1,277 @@
+
+        var packageName = path.basename(packageFileName).replace(path.extname(packageFileName), '');
+        var packageDir = path.join(path.dirname(packageFileName), packageName);
+
+        // makes a dir with the name of
+        // the OAM/ZIP package and decompress
+        // contents into it
+        fsextra.ensureDirSync(packageDir);
+
+        decompress(packageFileName, packageDir)
+          .then(function onDecompressed(filesFromPackage) {
+
+            validatePackage(packageDir, filesFromPackage,
+              function onValidationDone(err) {
+                if(!err) {
+                  // removes redundant root folder from
+                  // the OAM package file
+                  removePackageRoot(packageDir, filesFromPackage);
+
+                  // removes the oam package, not
+                  // needed in exported course
+                  fsextra.removeSync(packageFileName);
+
+                  unpackedCount++;
+
+                  if (unpackedCount === total) {
+                    onUnpackDone(null, filesFromPackage);
+                  }
+                }
+                else {
+                  logger.log('error', ERROR_UNPACKING, err);
+                  onUnpackDone(err, null);
+                }
+              });
+
+          })
+          .catch(function(err) {
+            logger.log('error', ERROR_UNPACKING, err);
+            onUnpackDone(err, null);
+          });
+      })();
+    }
+  },
+
+  /**
+   * Watch the course build folder where
+   * Adapt AT works to preview the course.
+   * Checks presence of OAM/ZIP files and unpacks
+   * them if any.
+   * @param watchingDir {string}
+   */
+  watchCourseOutputDir: function(watchingDir) {
+    var monitor = fsmonitor.watch(watchingDir, {
+      matches: function(relpath) {
+        return (
+          (relpath.match(/\.oam$/i) !== null)
+        );
+      },
+      // exclude not needed directories
+      // TODO: periodically review this list on new AT releases
+      excludes: function(relpath) {
+        return (
+          relpath.match(/^\.git$/i) !== null ||
+          relpath.match(/^\node_modules$/i) !== null ||
+          relpath.match(/^\grunt/i) !== null ||
+          relpath.match(/^\src/i) !== null
+        );
+      }
+    });
+
+    monitor.on('change', function onChange(changes) {
+      var packagesList = [];
+
+      for (var i in changes.addedFiles) {
+        packagesList.push(path.join(watchingDir, changes.addedFiles[i]));
+      }
+      for (var ii in changes.modifiedFiles) {
+        packagesList.push(path.join(watchingDir, changes.modifiedFiles[ii]));
+      }
+      //
+      // Monitor.on('change') also returns "changes.removedFiles"
+      // but we don't care about them since here we
+      // are working with the course preview only,
+      // and the output doesn't go into the course
+      // export downloaded by the user.
+      pluginInstance.unpack(packagesList, function onUnpacked(err) {
```

Comment:
Hi @fabiobeoni, 
Once the .oam file had been removed, fsmonitor.on('change') will trigger unpack() again. I think it's worth to consider removing that file from watchers, or perhaps in unpack() we have a checkpoint - something before running loopPackages() like:
`if (!fsextra.existsSync(packageFileName)&& (unpackedCount === total)) {
 onUnpackDone(null, filesFromPackage);
}`
Thanks again,
Cheers

Reply:
Totally agree. Thanks